server/sonar-web/src/main/js/components/common/MultiSelect.js

```diff
@@ -0,0 +1,220 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// @flow
+import React from 'react';
+import difference from 'lodash/difference';
+import MultiSelectOption from './MultiSelectOption';
+import { translate } from '../../helpers/l10n';
+
+type Props = {
+  selectedElements: Array<string>,
+  elements: Array<string>,
+  onSearch: (string) => void,
+  onSelect: (string) => void,
+  onUnselect: (string) => void
+};
+
+type State = {
+  query: string,
+  unselectedElements: Array<string>,
+  active: { idx: number, value: string }
+};
+
+export default class MultiSelect extends React.PureComponent {
```

Comment:
when selecting or deselecting an items, the selection should stay on the same **index**